docs/devel/luns.md

```diff
@@ -0,0 +1,82 @@
+----------------------------------------
+
+1. Destination (OCP cluster) reachability to the ISCSI storage, where the LUN is.
+2. RBAC permission to create `PersistentVolumes` by the controller.
+   - Will be added to the operator manifest.
+
+Risks and Mitigations
+----------------------------------------
+
+1. A VM that will be used while or after the migration with a direct LUN connected.
+   - A warning should be shown before starting the migration.
+   - Detaching the LUN from source VM, once the migration completes.
+   - (Optional) UI checkbox to the user, before starting the migration.
+   - The migration _must_ be a cold migration.
+2. An oVirt version with unsupported API. The [PR](https://github.com/oVirt/ovirt-engine/pull/559) provides LUN details is required.
+   - We may check for missing LUN details and skip the disk migration.
+   - (Optional) Have a mechanism to check the oVirt version for future work.
+3. Creating a `PersistentVolume` requires the file system type: `ext4` or `xfs` of the LUN, which we don't know.
+   - Might not be required for `Block` type.
+4. On non-local destination, the controller needs permission to create the `PersistentVolumes`.
+   - It may be enough to be added to the forklift-operator.
+
+Implementation
+----------------------------------------
+
+The controller gather data from source provider and saves them to the inventory.
+As part of the data gathering we will gather the LUN disks data. We have the VMs disk attachments where we follow to the disks.
+Within the disks API, if the _storage\_type_ is `LUN`, we will gather from the API endpoint `api/disks/<disk_id>` the size, address, IQN and the mapping of the LUN.
+Those will be found withing the `logical_unit` element of the `lun_storage`.
+On the migration plan a new steps will be added, creating a `PersistentVolume`(PV) and a `PersistentVolumeClaim`(PVC) per LUN disk.
+Both of them will contain the relevant data to use the LUN, the PVC will be bound the the PV by a unique label, `<vm_id>-<vm_name>-vol-<volume-number>`.
+When the VM specification created, the controller will add a LUN disk using the PVC that was created as [described by kubevirt](https://kubevirt.io/user-guide/virtual_machines/disks_and_volumes/#lun).
+
+Relevant validations will be performed to know we have the LUN details:
+
+- Allow `LUN` storage type on existing validation.
+- LUN details exist on the provider.
+- SCSI persistent reservation (exists), may required further warning when LUN is detected.
+- (Optional) Make sure the VM is in `down` on source provider(might be not required and be part of the execution plan).
+
+Execution Flow
+----------------------------------------
+
+1. The inventory will be filled based on the provider.
+   - On each disk, try to gather `lun_storage` data.
+2. Validation rules will be executed.
+3. Two steps will be added to the migration:
+   - `CreatePersistentVolumes`, which will create the relevant `PV` based on the `LUN` details.
```

Comment:
did you consider checking on the destination cluster if we have a storage provisioner that can prepare the PV for us first?

Reply:
no, can you elaborate? i did try on my cluster and i had no permission to do it out of the box.

Reply:
let's say that the user uses netapp on rhv and there's a netapp provisioner on the target ocp cluster - I'd expect to delegate the configuration of the lun to the netapp provisioner rather than posting such "static" PVs. posting PVs should really be done as a last resort, if there is no other way to get it

Reply:
We did talk more about it offline - currently the next step is to check if CSI can do the job and save us from creating the PV.
In that case, we will need to check the vendor of the LUN (provided in the API) - automating the storage class, or:
create mapping like the current SD mapping, just based on LUN to Storage Class with the relevant CSI to use.
Then, it may seem we really don't need much other than setting up the PVC with the mapped SC. i am working to check this flow.
Once i'll see it works, i'll refresh the FP.